src/generate-query.ts

```diff
@@ -608,13 +608,15 @@ function getArgsAndVars(
             schema
           )
 
-          if (typeof value !== 'object') {
-            variableValues[varName] = value
-          } else {
-            variableValues[varName] = merge.recursive(
-              value,
-              variableValues[varName]
-            )
+          if (value !== undefined) {
+            if (typeof value !== 'object') {
+              variableValues[varName] = value
```

Comment:
What I expected you to do was, instead of adding the `@listSize` directive default values here (or more properly, single default value), pass `variableValues` into `getListSizeDirectiveDefaultValue()`. That way, you can add multiple values into the `variableValues` as you iterate through all of slicing arguments if `requireOneSlicingArg` is set to `false`.